examples/03-ui-components/12-custom-ui/MUISideMenu.tsx

```diff
@@ -0,0 +1,97 @@
+import { SideMenuProps } from "@blocknote/react";
+import { Delete, DragIndicator } from "@mui/icons-material";
+import {
+  Box,
+  ListItemIcon,
+  ListItemText,
+  Menu,
+  MenuItem,
+  Typography,
+} from "@mui/material";
+import { useMemo, useState } from "react";
+
+import { TextBlockSchema } from "./schema";
+
+// Custom component to replace the default Block Side Menu.
+export function MUISideMenu(props: SideMenuProps<TextBlockSchema>) {
+  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+
+  // Since the side menu is positioned by the top-left corner of a block, we\
+  // manually set its height based on the hovered block so that it's vertically
+  // centered.
+  const sideMenuHeight = useMemo(() => {
```

Comment:
maybe extract this into a reusable hook?

Reply:
I don't think it makes sense here, since in the `react` package we return `data-*` which are used in CSS, whereas in the example we return numbers which are used in the `sx` prop. If MUI was styled using vanilla CSS it would make sense to use the same hook for both, but the convention is to style your components with the `sx` prop instead.